src/giskardpy/python_interface.py

```diff
@@ -50,15 +54,86 @@ def get_root(self):
+    def get_robot_links(self):
+        """
+        Returns a list of the robots links
+        :rtype: dict
+        """
+        return self.robot_urdf.get_link_names()
+
+    def get_joint_states(self, topic=u'joint_states', timeout=1):
+        """
+        Returns a dictionary of all joints (key) and their position (value)
+        :param topic: joint_state topic
+        :param timeout: duration to wait for JointState msg
+        :return: OrderedDict[str, float]
+        """
+        try:
+            msg = rospy.wait_for_message(topic, JointState, rospy.Duration(timeout))
+            return to_joint_state_position_dict(msg)
+        except rospy.ROSException:
+            rospy.logwarn("get_joint_states: wait_for_message timeout")
+            return {}
+
+    def set_cart_goal_wstep(self, root_link, tip_link, goal_pose, base_tip_rotation=None,
+                            max_linear_velocity=None, max_angular_velocity=None, weight=None, step=None,
+                            base_pose=None):
+        """
+        This goal will use the kinematic chain between root and tip link to move tip link into the goal pose. Adds an offset
+        depending on the goal. It will also execute the goal.
+        :param root_link: name of the root link of the kin chain
+        :type root_link: str
+        :param tip_link: name of the tip link of the kin chain
+        :type tip_link: str
+        :param goal_pose: the goal pose
+        :type goal_pose: PoseStamped
+        :param base_tip_rotation: the rotation of the gripper
+        :type base_tip_rotation: tf_quaternion [x, y, z, w]
+        :param max_linear_velocity: m/s, default 0.1
+        :type max_linear_velocity: float
+        :param max_angular_velocity: rad/s, default 0.5
+        :type max_angular_velocity: float
+        :param weight: default WEIGHT_ABOVE_CA
+        :type weight: float
+        :param step: Distance of a potential step in front of the object
+        :type step: float (meter)
+        :param base_pose: the current pose of the robot
+        :type base_pose: PoseStamped
+        """
+        rotation = goal_pose.pose.orientation
+        if base_pose and base_tip_rotation:
+            hsr_rotation = to_tf_quaternion(base_pose.pose.orientation)
+            rotation = Quaternion(*quaternion_multiply(hsr_rotation, base_tip_rotation))
+
+        if base_pose and step:
+            step_pose = PoseStamped()
+            step_pose.header.frame_id = goal_pose.header.frame_id
+            step_pose.header.stamp = rospy.Time.now()
+            step_pose.pose.position = calculate_waypoint2D(goal_pose.pose.position, base_pose.pose.position, step)
+            step_pose.pose.orientation = rotation
+            rospy.loginfo("step_pose: {}".format(step_pose))
+            # Move to the defined step
+            self.set_cart_goal(root_link, tip_link, step_pose, max_linear_velocity, max_angular_velocity, weight)
+            self.plan_and_execute(wait=True)
```

Comment:
```plan_and_execute``` in a set goal method is inconsistent behavior with the rest of the interface. Instead I would prefer a new function called e.g. ```set_cart_gregrasp_goal```, that only computes and sets this goal. It can then be used in combination with normal cart goals, is the user wants to.

Reply:
I have looked into the code in SUTURO_Manipulation and noticed, that we aren't using this method. We are already splitting the motion in 2 separate ones in that specific code. The first step is calculated with `calculate_waypoint2D()`. It is already in the intended way, which u mentioned above. I think it is understandable, that this method is obsolete and therefore will be removed.

Reply:
@ichumuh What do you think?

Reply:
Alright, remove it.

Reply:
@ichumuh  Is removed